cpp/CMakeLists.txt

```diff
@@ -14,713 +14,512 @@
+option(AUTO_DETECT_CUDA_ARCHITECTURES "Auto-detect CUDA architectures to build from the GPUs on the current system" OFF)
+option(DISABLE_DEPRECATION_WARNING "Disable warnings generated from deprecated declarations." OFF)
+# Option to enable line info in CUDA device compilation to allow introspection when profiling / memchecking
+option(CUDA_ENABLE_LINEINFO "Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler" OFF)
 # cudart can be statically linked or dynamically linked. The python ecosystem wants dynamic linking
-
 option(CUDA_STATIC_RUNTIME "Statically link the CUDA runtime" OFF)
 
-if(CUDA_STATIC_RUNTIME)
-    message(STATUS "Enabling static linking of cudart")
-    set(CUDART_LIBRARY "cudart_static")
-else()
-    set(CUDART_LIBRARY "cudart")
-endif(CUDA_STATIC_RUNTIME)
+message(STATUS "CUDF: Build with NVTX support: ${USE_NVTX}")
+message(STATUS "CUDF: Configure CMake to build tests: ${BUILD_TESTS}")
+message(STATUS "CUDF: Configure CMake to build (google) benchmarks: ${BUILD_BENCHMARKS}")
+message(STATUS "CUDF: Build cuDF shared libraries: ${BUILD_SHARED_LIBS}")
+message(STATUS "CUDF: Use a file cache for JIT compiled kernels: ${JITIFY_USE_CACHE}")
+message(STATUS "CUDF: Build and statically link Arrow libraries: ${ARROW_STATIC_LIB}")
+message(STATUS "CUDF: Build with per-thread default stream: ${PER_THREAD_DEFAULT_STREAM}")
+message(STATUS "CUDF: Auto-detect CUDA architectures to build from the GPUs on the current system: ${AUTO_DETECT_CUDA_ARCHITECTURES}")
+message(STATUS "CUDF: Disable warnings generated from deprecated declarations: ${DISABLE_DEPRECATION_WARNING}")
+message(STATUS "CUDF: Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler: ${CUDA_ENABLE_LINEINFO}")
+message(STATUS "CUDF: Statically link the CUDA runtime: ${CUDA_STATIC_RUNTIME}")
 
-###################################################################################################
-# - cmake modules ---------------------------------------------------------------------------------
+# Set a default build type if none was specified
+set(DEFAULT_BUILD_TYPE "Release")
+set(CUDF_BUILD_TESTS ${BUILD_TESTS})
+set(CUDF_BUILD_BENCHMARKS ${BUILD_BENCHMARKS})
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/" ${CMAKE_MODULE_PATH})
+set(CUDF_CXX_FLAGS "")
+set(CUDF_CUDA_FLAGS "")
+set(CUDF_CXX_DEFINITIONS "")
+set(CUDF_CUDA_DEFINITIONS "")
 
-include(FeatureSummary)
-include(CheckIncludeFiles)
-include(CheckLibraryExists)
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "CUDF: Setting build type to '${DEFAULT_BUILD_TYPE}' since none specified.")
+    set(CMAKE_BUILD_TYPE "${DEFAULT_BUILD_TYPE}" CACHE STRING  "Choose the type of build." FORCE)
+    # Set the possible values of build type for cmake-gui
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
```

Comment:
I think the previous comment was lost, but can we do something like `string(TOUPPER $CMAKE_BUILD_TYPE CMAKE_BUILD_TYPE)` here to convert it to upper case so that it does't matter whether someone does `release` vs `Release` vs `RELEASE`?

Reply:
You mean for setting the `CMAKE_BUILD_TYPE`? Doesn't CMake require the string is camel-case?

Reply:
I did not realize this is a standard CMake thing, *facepalm*. This is fine then, ignore me! 😄